pluma/pluma-app-activatable.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * pluma-app-activatable.h
+ * This file is part of pluma
+ *
+ * Copyright (C) 2010 - Steve Frécinaux
+ * Copyright (C) 2010 - Jesse van den Kieboom
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Library General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+#ifndef __PLUMA_APP_ACTIVATABLE_H__
+#define __PLUMA_APP_ACTIVATABLE_H__
+
+#include <glib-object.h>
+
+G_BEGIN_DECLS
+
+/*
+ * Type checking and casting macros
+ */
+#define PLUMA_TYPE_APP_ACTIVATABLE           (pluma_app_activatable_get_type ())
+#define PLUMA_APP_ACTIVATABLE(obj)           (G_TYPE_CHECK_INSTANCE_CAST ((obj),    PLUMA_TYPE_APP_ACTIVATABLE, PlumaAppActivatable))
+#define PLUMA_APP_ACTIVATABLE_IFACE(obj)     (G_TYPE_CHECK_CLASS_CAST ((obj),       PLUMA_TYPE_APP_ACTIVATABLE, PlumaAppActivatableInterface))
+#define PLUMA_IS_APP_ACTIVATABLE(obj)        (G_TYPE_CHECK_INSTANCE_TYPE ((obj),    PLUMA_TYPE_APP_ACTIVATABLE))
+#define PLUMA_APP_ACTIVATABLE_GET_IFACE(obj) (G_TYPE_INSTANCE_GET_INTERFACE ((obj), PLUMA_TYPE_APP_ACTIVATABLE, PlumaAppActivatableInterface))
+
+typedef struct _PlumaAppActivatable           PlumaAppActivatable; /* dummy typedef */
+typedef struct _PlumaAppActivatableInterface  PlumaAppActivatableInterface;
```

Comment:
You can boilerplate the code:
```diff
diff --git a/pluma/pluma-app-activatable.h b/pluma/pluma-app-activatable.h
index 36072ad..4ddc3c2 100644
--- a/pluma/pluma-app-activatable.h
+++ b/pluma/pluma-app-activatable.h
@@ -26,17 +26,8 @@
 
 G_BEGIN_DECLS
 
-/*
- * Type checking and casting macros
- */
-#define PLUMA_TYPE_APP_ACTIVATABLE           (pluma_app_activatable_get_type ())
-#define PLUMA_APP_ACTIVATABLE(obj)           (G_TYPE_CHECK_INSTANCE_CAST ((obj),    PLUMA_TYPE_APP_ACTIVATABLE, PlumaAppActivatable))
-#define PLUMA_APP_ACTIVATABLE_IFACE(obj)     (G_TYPE_CHECK_CLASS_CAST ((obj),       PLUMA_TYPE_APP_ACTIVATABLE, PlumaAppActivatableInterface))
-#define PLUMA_IS_APP_ACTIVATABLE(obj)        (G_TYPE_CHECK_INSTANCE_TYPE ((obj),    PLUMA_TYPE_APP_ACTIVATABLE))
-#define PLUMA_APP_ACTIVATABLE_GET_IFACE(obj) (G_TYPE_INSTANCE_GET_INTERFACE ((obj), PLUMA_TYPE_APP_ACTIVATABLE, PlumaAppActivatableInterface))
-
-typedef struct _PlumaAppActivatable           PlumaAppActivatable; /* dummy typedef */
-typedef struct _PlumaAppActivatableInterface  PlumaAppActivatableInterface;
+#define PLUMA_TYPE_APP_ACTIVATABLE (pluma_app_activatable_get_type ())
+G_DECLARE_INTERFACE (PlumaAppActivatable, pluma_app_activatable, PLUMA, APP_ACTIVATABLE, GObject)
 
 struct _PlumaAppActivatableInterface
 {
@@ -50,7 +41,6 @@ struct _PlumaAppActivatableInterface
 /*
  * Public methods
  */
-GType    pluma_app_activatable_get_type     (void)  G_GNUC_CONST;
 
 void     pluma_app_activatable_activate     (PlumaAppActivatable *activatable);
 void     pluma_app_activatable_deactivate   (PlumaAppActivatable *activatable);
diff --git a/pluma/pluma-view-activatable.h b/pluma/pluma-view-activatable.h
index 5a6635b..acc6402 100644
--- a/pluma/pluma-view-activatable.h
+++ b/pluma/pluma-view-activatable.h
@@ -25,17 +25,8 @@
 
 G_BEGIN_DECLS
 
-/*
- * Type checking and casting macros
- */
-#define PLUMA_TYPE_VIEW_ACTIVATABLE             (pluma_view_activatable_get_type ())
-#define PLUMA_VIEW_ACTIVATABLE(obj)             (G_TYPE_CHECK_INSTANCE_CAST ((obj),    PLUMA_TYPE_VIEW_ACTIVATABLE, PlumaViewActivatable))
-#define PLUMA_VIEW_ACTIVATABLE_IFACE(obj)       (G_TYPE_CHECK_CLASS_CAST ((obj),       PLUMA_TYPE_VIEW_ACTIVATABLE, PlumaViewActivatableInterface))
-#define PLUMA_IS_VIEW_ACTIVATABLE(obj)          (G_TYPE_CHECK_INSTANCE_TYPE ((obj),    PLUMA_TYPE_VIEW_ACTIVATABLE))
-#define PLUMA_VIEW_ACTIVATABLE_GET_IFACE(obj)   (G_TYPE_INSTANCE_GET_INTERFACE ((obj), PLUMA_TYPE_VIEW_ACTIVATABLE, PlumaViewActivatableInterface))
-
-typedef struct _PlumaViewActivatable           PlumaViewActivatable; /* dummy typedef */
-typedef struct _PlumaViewActivatableInterface  PlumaViewActivatableInterface;
+#define PLUMA_TYPE_VIEW_ACTIVATABLE (pluma_view_activatable_get_type ())
+G_DECLARE_INTERFACE (PlumaViewActivatable, pluma_view_activatable, PLUMA, VIEW_ACTIVATABLE, GObject)
 
 struct _PlumaViewActivatableInterface
 {
@@ -49,7 +40,6 @@ struct _PlumaViewActivatableInterface
 /*
  * Public methods
  */
-GType    pluma_view_activatable_get_type    (void)  G_GNUC_CONST;
 
 void     pluma_view_activatable_activate    (PlumaViewActivatable *activatable);
 void     pluma_view_activatable_deactivate  (PlumaViewActivatable *activatable);
diff --git a/pluma/pluma-window-activatable.h b/pluma/pluma-window-activatable.h
index 9646507..cb284bf 100644
--- a/pluma/pluma-window-activatable.h
+++ b/pluma/pluma-window-activatable.h
@@ -25,17 +25,8 @@
 
 G_BEGIN_DECLS
 
-/*
- * Type checking and casting macros
- */
-#define PLUMA_TYPE_WINDOW_ACTIVATABLE           (pluma_window_activatable_get_type ())
-#define PLUMA_WINDOW_ACTIVATABLE(obj)           (G_TYPE_CHECK_INSTANCE_CAST ((obj),    PLUMA_TYPE_WINDOW_ACTIVATABLE, PlumaWindowActivatable))
-#define PLUMA_WINDOW_ACTIVATABLE_IFACE(obj)     (G_TYPE_CHECK_CLASS_CAST ((obj),       PLUMA_TYPE_WINDOW_ACTIVATABLE, PlumaWindowActivatableInterface))
-#define PLUMA_IS_WINDOW_ACTIVATABLE(obj)        (G_TYPE_CHECK_INSTANCE_TYPE ((obj),    PLUMA_TYPE_WINDOW_ACTIVATABLE))
-#define PLUMA_WINDOW_ACTIVATABLE_GET_IFACE(obj) (G_TYPE_INSTANCE_GET_INTERFACE ((obj), PLUMA_TYPE_WINDOW_ACTIVATABLE, PlumaWindowActivatableInterface))
-
-typedef struct _PlumaWindowActivatable           PlumaWindowActivatable; /* dummy typedef */
-typedef struct _PlumaWindowActivatableInterface  PlumaWindowActivatableInterface;
+#define PLUMA_TYPE_WINDOW_ACTIVATABLE (pluma_window_activatable_get_type ())
+G_DECLARE_INTERFACE (PlumaWindowActivatable, pluma_window_activatable, PLUMA, WINDOW_ACTIVATABLE, GObject)
 
 struct _PlumaWindowActivatableInterface
 {
@@ -50,7 +41,6 @@ struct _PlumaWindowActivatableInterface
 /*
  * Public methods
  */
-GType    pluma_window_activatable_get_type      (void)  G_GNUC_CONST;
 
 void     pluma_window_activatable_activate      (PlumaWindowActivatable *activatable);
 void     pluma_window_activatable_deactivate    (PlumaWindowActivatable *activatable);
```
Ref: https://developer.gnome.org/gobject/stable/howto-interface.html

Reply:
thanks, done